app/main.ts

```diff
@@ -28,18 +35,39 @@ async function start(): Promise<void> {
       },
     },
   })
-
+  // This request comes from the renderer
   ipcMain.handle(`browse-for-site`, async () => {
     console.log(`Browsing for site`)
     const { canceled, filePaths } = await dialog.showOpenDialog({
       properties: [`openDirectory`],
     })
-    if (canceled) {
+    if (canceled || !filePaths?.length) {
       return undefined
     }
-    console.log({ filePaths })
-    // TODO: check it's a Gatsby site here
-    return filePaths
+
+    const path = filePaths[0]
+    // i.e. actually installed in node_modules
+    if (await hasGatsbyInstalled(path)) {
+      const packageJson = await loadPackageJson(path)
+      return { packageJson, path }
+    }
+
+    try {
+      const packageJson = await loadPackageJson(path)
```

Comment:
Considering you're going to fetch the package.json, regardless, wouldn't it be more efficient to fetch it first and check if it exists? Perf wise it's not likely to make a difference but the logic looks a little weird as in both cases you'll be fetching it, anyways.